src/mmw/js/src/modeling/controllers.js

```diff
@@ -139,37 +133,31 @@ var ModelingController = {
                     lock.resolve();
                 }
 
-                lock.done(function() {
-                    project.on('change:id', updateUrl);
-                    initScenarioEvents(project);
-                    initViews(project);
-                    if (App.projectNumber) {
-                        updateUrl();
-                    }
-                });
+                finishProjectSetup(lock, project);
             }
         }
 
         App.state.set('current_page_title', 'Model');
     },
 
+    makeNewProject: function(modelPackage) {
+        var project;
+        var lock = $.Deferred();
+        project = makeProject(lock, modelPackage);
+
+        App.currentProject = project;
+
+        setupNewProjectScenarios(project);
+        finishProjectSetup(lock, project);
+        updateUrl();
+    },
+
     projectCleanUp: function() {
-        App.rootView.hideCollapsable();
-        if (App.currentProject) {
-            var scenarios = App.currentProject.get('scenarios');
-
-            App.currentProject.off('change:id', updateUrl);
-            scenarios.off('change:activeScenario change:id', updateScenario);
-            // App.projectNumber holds the number of the project that was
-            // in use when the user left the `/project` page.  The intent
-            // is to allow the same project to be returned-to via the UI
-            // arrow buttons (see issue #690).
-            App.projectNumber = scenarios.at(0).get('project');
-        }
+        projectCleanUp();
+    },
 
-        App.getMapView().updateModifications(null);
-        App.rootView.subHeaderRegion.empty();
-        App.rootView.sidebarRegion.empty();
+    makeNewProjectCleanUp: function() {
```

Comment:
The function name is constructed programmatically, so it doesn't show up if you search the source code.